src/stack.rs

```diff
@@ -1,14 +1,117 @@
-// Alex Ozdemir <aozdemir@hmc.edu> // <- Your name should replace this line!
+// Dan Obermiller <dobermiller16@cmc.edu> // <- Your name should replace this line!
 // Starter code for HMC's MemorySafe, week 1
 //
 // The implementation of SinglyLinkedList
 
+use std::mem;
 use Stack;
 
+struct Node<T> {
+    pointer: Option<Box<Node<T>>>,
+    data: T,
+}
+
 pub struct SinglyLinkedList<T> {
-    data: T // Probably not what you want
+    head: Option<Box<Node<T>>>,
+    size: usize,
 }
 
 impl<T: Eq> Stack<T> for SinglyLinkedList<T> {
-    // ????
+    fn new() -> Self {
+        SinglyLinkedList{head: None, size: 0}
+    }
+
+    fn push_front(&mut self, item: T) {
+        let old = mem::replace(&mut self.head, None);
+        self.head = Some(Box::new(Node { pointer: old, data: item }));
+        // let mut new_head = Node { pointer: None, data: item };
+        // mem::swap(&mut new_head.pointer, &mut self.head);
+        // self.head = Some(Box::new(new_head));
+        self.size += 1;
+    }
+
+    fn pop_front(&mut self) -> Option<T> {
+        self.head.take().map(|mut n| {
+            assert!(self.size > 0);
```

Comment:
Cool tidbit: `debug_assert!` is similar but is removed when optimization is enabled.